ecs-agent/acs/session/payload_responder.go

```diff
@@ -0,0 +1,162 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+)
+
+const (
+	PayloadMessageName = "PayloadMessage"
+)
+
+type TaskMessageHandler interface {
+	ProcessTaskMessage(message *ecsacs.PayloadMessage,
```

Comment:
With changing the handler type name we still usages `ecsacs.PayloadMessage` as parameter makes it look a little bit inconsistent for me. If we are able to create a "taskMessage" interface and let `PayloadMessage` implement it will look nicer.

Reply:
See this comment: https://github.com/aws/amazon-ecs-agent/pull/3840#discussion_r1288885624